### PR TITLE
Upload inline images in Froala

### LINKED ISSUE #4210: Make it possible to add an inline image in Froala

### PR DESCRIPTION
**Pull request by adishy**
### Status
**READY**

### Description
* It should be possible for instructors and students to upload an inline image in the Froala editor. Currently uploading images does not work and is disabled. Now, uploading images is allowed and is uploaded to a new directory in the uploads folder (files/uploads/froala_uploads) and can be inserted as inline images in the editor. Text comments within submissions by students are  exported as pdfs and have inline image uploads attached and also preserve inline tables and formatting in the text comment.

### Related PRs
branch | PR
------ | ------
image-button | #4235 

### Gem dependencies
- ADDED [wkhtmltopdf-binary](https://github.com/zakird/wkhtmltopdf_binary_gem)
- ADDED [wicked_pdf](https://github.com/mileszs/wicked_pdf)

### Migrations
NO

### Steps to Test or Reproduce
* Uploading inline images
1. As an instructor or student access a page that has the Froala editor and allows images to be uploaded. Click the image button in the editor and upload an image. 
2. The image that is uploaded should be inserted as an inline image in the editor and it should be possible to view and edit the uploaded image even after navigating away and then back to the page where the image was uploaded in.

* Exporting text comments within submissions by students
1. As a student, submit an assignment with a text comments that has an inline image upload, an inline table and other formatting (any combination of these).
2. As an instructor, download all submissions for the assignment that the submission was made in. The downloaded archive should have text submissions exported as pdfs with formatting from the text comment as well as any inline images. Any comments that contain only text should also be exported as pdfs in the archive.

### Impacted Areas in Application
* Uploading inline images in the Froala editor
* Exporting text comments in submissions by students
* services/creates_submissions_export/create_submission_files.rb 

======================
Closes #4210

### CODE REVIEW

**Comment by adishy** [comment]
* The following pages from the Froala documentation website were used:
   * https://www.froala.com/wysiwyg-editor/v2.0/docs/framework-plugins/rails
   * https://www.froala.com/wysiwyg-editor/docs/server/ruby/image-upload
*  Uploads are stored to the files/uploads/froala_uploads directory and are uploaded and viewed using the Froala uploads controller (controllers/froala_uploads_controller.rb)

**Comment by ebarroso7** [comment]
@adishy I also just tried to copy the course that I did this testing in, and it did not copy over the inline images

File: spec/services/creates_submissions_export/create_submission_files_spec.rb [hunk 1]
```diff
@@ -34,27 +34,6 @@
       end
     end
 
-    context "for teams" do
```
[the remark this answers is not in the document]
Comment (adishy) [reply spec/services/creates_submissions_export/create_submission_files_spec.rb]: These spec tests checks for a text comment attribute in each submission, which is now changed to be a PDF file.

File: app/assets/javascripts/utilities.js [hunk 2]
```diff
@@ -31,18 +31,37 @@ $( "#tabs" ).tabs({
   key: 'RD4H4B12B7iB6E5C3A4I2I3C8B6B5A4C-11NGNe1IODMGYNSFKV==',
   inlineMode: false,
   heightMin: 200,
+  imageUploadURL: '/api/upload_froala_images',
+  events: {
```
[the remark this answers is not in the document]
Comment (adishy) [reply app/assets/javascripts/utilities.js]: Nope, that was probably left over from testing, will remove it!

**Pull request merged**